src/pages/AdvancedSearch/SearchForm/SearchForm.tsx

```diff
@@ -0,0 +1,73 @@
+import { Button, Col, Row, Textarea } from "@canonical/react-components";
+import type { FormikProps } from "formik";
+import { Field, Form, Formik } from "formik";
+import { useEffect, useRef } from "react";
+
+import { useQueryParams } from "hooks/useQueryParams";
+import {
+  getControllerConnection,
+  getWSControllerURL,
+} from "store/general/selectors";
+import { actions as jujuActions } from "store/juju";
+import { useAppDispatch, useAppSelector } from "store/store";
+
+type Fields = {
+  query: string;
+};
+
+const SearchForm = (): JSX.Element => {
+  const formikRef = useRef<FormikProps<Fields>>(null);
+  const dispatch = useAppDispatch();
+  const wsControllerURL = useAppSelector(getWSControllerURL);
+  const hasControllerConnection = useAppSelector((state) =>
+    getControllerConnection(state, wsControllerURL)
+  );
+  const [queryParams, setQueryParams] = useQueryParams<{ q: string }>({
+    q: "",
+  });
+  const query = decodeURIComponent(queryParams.q);
+
+  useEffect(() => {
+    if (query && hasControllerConnection && wsControllerURL) {
+      dispatch(jujuActions.fetchCrossModelQuery({ query, wsControllerURL }));
+    }
+  }, [dispatch, hasControllerConnection, query, wsControllerURL]);
+
+  return (
+    <Formik<Fields>
+      initialValues={{
+        query,
+      }}
+      innerRef={formikRef}
+      onSubmit={(values) => {
+        const { query } = values;
+        if (query) {
+          setQueryParams({ q: encodeURIComponent(query) });
```

Comment:
This is just a general thought that crossed my mind now. If we set the query params like this throughout our application, aren't we open to cross-site scripting attacks? Or is this not really a concern for us? I think that this issue can be easily remedied by using something like `DOMPurify.sanitize()` in `useQueryParams`.

Reply:
I don't think this has been a big issue until now, as I think in almost every other case the values have been internal (e.g. the panel) or from the API (e.g. model uuid) instead of user provided (except for the app search in model details).

I think you're right, we should sanitise the values in `useQueryParams`. I'll add a task to do that.